ui/app/AppLayouts/Onboarding/views/SeedPhraseInputView.qml

```diff
@@ -64,6 +65,40 @@ OnboardingBasePage {
             z: 100000
             model: switchTabBar.currentItem.text.substring(0,2) === "12" ? 12 :
                    switchTabBar.currentItem.text.substring(0,2) === "18" ? 20 : 24
+
+            function addWord(pos, word, ignoreGoingNext) {
+                root.mnemonicInput.push({pos: pos, seed: word.replace(/\s/g, '')});
+                for (var j = 0; j < mnemonicInput.length; j++) {
+                    if (mnemonicInput[j].pos === pos && mnemonicInput[j].seed !== word) {
+                        mnemonicInput[j].seed = word;
+                        break;
+                    }
+                }
+                //remove duplicates
+                var valueArr = mnemonicInput.map(function(item){ return item.pos });
+                var isDuplicate = valueArr.some(function(item, idx){
+                    if (valueArr.indexOf(item) !== idx) {
+                        root.mnemonicInput.splice(idx, 1);
+                    }
+                    return valueArr.indexOf(item) !== idx
+                });
+                if (!ignoreGoingNext) {
+                    for (var i = !grid.atXBeginning ? 12 : 0; i < grid.count; i++) {
+                        if (parseInt(grid.itemAtIndex(i).leftComponentText) === (parseInt(pos)+1)) {
+                            grid.currentIndex = grid.itemAtIndex(i).itemIndex;
+                            grid.itemAtIndex(i).textEdit.input.edit.forceActiveFocus();
+                            if (grid.currentIndex === 12) {
+                                grid.positionViewAtEnd();
+                                if (grid.count === 20) {
+                                    grid.contentX = 1500;
+                                }
+                            }
+                        }
```

Comment:
Maybe unroll this spaghetti like this?
```js
if (parseInt(grid.itemAtIndex(i).leftComponentText) !== (parseInt(pos)+1))
    continue;

grid.currentIndex = grid.itemAtIndex(i).itemIndex;
grid.itemAtIndex(i).textEdit.input.edit.forceActiveFocus();

if (grid.currentIndex !== 12)
    continue;

grid.positionViewAtEnd();

if (grid.count === 20) 
    grid.contentX = 1500;
```

Reply:
This will be gone soon https://github.com/status-im/status-desktop/pull/5694